cluster/kube-util.sh

```diff
@@ -43,22 +43,29 @@ function set-federation-zone-vars {
     echo "Setting zone vars to: $OVERRIDE_CONTEXT"
     if [[ "$KUBERNETES_PROVIDER" == "gce"  ]];then
 
-	export KUBE_GCE_ZONE="$zone"
-	# gcloud has a 61 character limit, and for firewall rules this
-	# prefix gets appended to itself, with some extra information
-	# need tot keep it short
-	export KUBE_GCE_INSTANCE_PREFIX="${USER}-${zone}"
+        export KUBE_GCE_ZONE="$zone"
+        # gcloud has a 61 character limit, and for firewall rules this
+        # prefix gets appended to itself, with some extra information
+        # need tot keep it short
+        export KUBE_GCE_INSTANCE_PREFIX="${USER}-${zone}"
 
     elif [[ "$KUBERNETES_PROVIDER" == "gke"  ]];then
 
-	export CLUSTER_NAME="${USER}-${zone}"
+        export CLUSTER_NAME="${USER}-${zone}"
 
     elif [[ "$KUBERNETES_PROVIDER" == "aws"  ]];then
 
-	export KUBE_AWS_ZONE="$zone"
-	export KUBE_AWS_INSTANCE_PREFIX="${USER}-${zone}"
+        export KUBE_AWS_ZONE="$zone"
+        export KUBE_AWS_INSTANCE_PREFIX="${USER}-${zone}"
+
+        # WARNING: This is hack
+        # After KUBE_AWS_INSTANCE_PREFIX is changed,
+        # we need to make sure the config-xxx.sh file is
+        # re-sourced so the change propogates to dependent computed values
+        # (eg: MASTER_SG_NAME, NODE_SG_NAME, etc)
+        source "${KUBE_ROOT}/cluster/aws/util.sh"
```

Comment:
I honestly can't say for sure, I'd have to re-read that util file. As happened here, it's stil possible we're stomping some resource definition by launching multiple clusters and just not noticing :-1:  

IIRC, GCE doesn't do cross-zone namespacing with the components we use, so we get away with conflicting names across our cluster set. We require 1 or 0 clusters per zone (for all providers), so it works out in the GCE case.